packages/forms/src/index.tsx

```diff
@@ -150,68 +152,249 @@ const useErrorStyles = ({
+> = {
+  // valueAsBoolean is commented out as r-h-f does not currently support
+  // setValueAs functionality for checkboxes.  May investigate future
+  // integration
+  /*  valueAsBoolean: {
+    // r-h-f returns a boolean if a checkBox type, but also handle string case in case valueAsBoolean is used
+    base: (val: boolean | string): boolean => !!val,
+    emptyAsNull: (val: boolean | string): boolean | null => (val ? true : null),
+    emptyAsUndefined: (val: boolean | string): boolean | undefined =>
+      val ? true : undefined,
+  },*/
+  valueAsDate: {
+    emptyAsNull: (val: string): Date | null =>
+      isValueEmpty(val) ? null : new Date(val),
+    emptyAsUndefined: (val: string): Date | undefined =>
+      isValueEmpty(val) ? undefined : new Date(val),
+    emptyAsString: (val: string): Date | '' =>
+      isValueEmpty(val) ? '' : new Date(val),
+    emptyAsZero: (val: string): Date | 0 =>
+      isValueEmpty(val) ? 0 : new Date(val),
+  },
+  valueAsJSON: {
+    emptyAsNull: (val: string) => {
+      if (isValueEmpty(val)) {
+        return null
+      }
+      try {
+        return JSON.parse(val)
+      } catch (e) {
+        return NaN // represents invalid JSON parse to JSONValidation function
+      }
+    },
+    emptyAsString: (val: string) => {
+      if (isValueEmpty(val)) {
+        return ''
+      }
+      try {
+        return JSON.parse(val)
+      } catch (e) {
+        return NaN // represents invalid JSON parse to JSONValidation function
+      }
+    },
+    emptyAsUndefined: (val: string) => {
+      if (isValueEmpty(val)) {
+        return undefined
+      }
+      try {
+        return JSON.parse(val)
+      } catch (e) {
+        return NaN // represents invalid JSON parse to JSONValidation function
+      }
+    },
+    emptyAsZero: (val: string) => {
+      if (isValueEmpty(val)) {
+        return 0
+      }
+      try {
+        return JSON.parse(val)
+      } catch (e) {
+        return NaN // represents invalid JSON parse to JSONValidation function
+      }
+    },
+  },
+  valueAsNumber: {
+    emptyAsNull: (val: string): number | null =>
+      isValueEmpty(val) ? null : +val,
+    emptyAsUndefined: (val: string): number | undefined =>
+      isValueEmpty(val) ? undefined : +val,
+    emptyAsNaN: (val: string): number | typeof NaN =>
+      isValueEmpty(val) ? NaN : +val,
+    emptyAsString: (val: string): number | '' =>
+      isValueEmpty(val) ? '' : +val,
+    emptyAsZero: (val: string): number => (isValueEmpty(val) ? 0 : +val),
+  },
+  valueAsString: {
+    emptyAsNull: (val: string) => (isValueEmpty(val) ? null : val),
+    emptyAsUndefined: (val: string) => (isValueEmpty(val) ? undefined : val),
+    emptyAsString: (val: string): string => (isValueEmpty(val) ? '' : val),
+    emptyAsZero: (val: string): string | number =>
+      isValueEmpty(val) ? 0 : val,
   },
 }
 
-const JSONValidation = (value: Record<string, unknown> | null) => value !== null
+const getSetValueAsFcn = (
```

Comment:
```suggestion
const getSetValueAsFn = (
```

Super nitpicky, but we shorten "Function" to just "Fn" in other parts of the code